src/SpinWaveTheory/SpinWaveTheory.jl

```diff
@@ -75,78 +77,79 @@ function generate_local_sun_gens(sys :: System)
     Q_mat_N[4] = s_mat_N[1] * s_mat_N[2] + s_mat_N[2] * s_mat_N[1]
     Q_mat_N[5] = √3 * s_mat_N[3] * s_mat_N[3] - 1/√3 * S * (S+1) * I
 
-    if sys.mode == :SUN
-        s_mat_N = spin_matrices(N)
-
-        s̃_mat = Array{ComplexF64, 4}(undef, N, N, 3, Nₘ)
-        T̃_mat = Array{ComplexF64, 3}(undef, N, N, Nₘ)
-        Q̃_mat = zeros(ComplexF64, N, N, 5, Nₘ)
+    U_mat = Matrix{ComplexF64}(undef, N, N)
 
-        U_mat = Matrix{ComplexF64}(undef, N, N)
+    s̃_mat = Array{ComplexF64, 4}(undef, N, N, 3, Nₘ)
+    T̃_mat = Array{ComplexF64, 3}(undef, N, N, Nₘ)
+    Q̃_mat = zeros(ComplexF64, N, N, 5, Nₘ)
 
-        for atom = 1:Nₘ
-            U_mat[:, 1] = sys.coherents[1, 1, 1, atom]
-            U_mat[:, 2:N] = nullspace(U_mat[:, 1]')
-            @assert isapprox(U_mat * U_mat', I) "rotation matrix from (global frame to local frame) not unitary"
-            for μ = 1:3
-                s̃_mat[:, :, μ, atom] = Hermitian(U_mat' * s_mat_N[μ] * U_mat)
-            end
-            for ν = 1:5
-                Q̃_mat[:, :, ν, atom] = Hermitian(U_mat' * Q_mat_N[ν] * U_mat)
-            end
-            T̃_mat[:, :, atom] = Hermitian(U_mat' * sys.interactions_union[atom].aniso.matrep * U_mat)
+    for atom = 1:Nₘ
+        U_mat[:, 1] = sys.coherents[1, 1, 1, atom]
+        U_mat[:, 2:N] = nullspace(U_mat[:, 1]')
+        @assert isapprox(U_mat * U_mat', I) "rotation matrix from (global frame to local frame) not unitary"
```

Comment:
Is there any reason this would ever fail, assuming that there is nothing wrong with the original coherent state? (I am wondering if we can remove the assertion test.)

Reply:
I think we can get rid of this assertion. Thanks!